Userland/Services/ChessEngine/MCTSTree.cpp

```diff
@@ -117,22 +130,37 @@ void MCTSTree::do_round()
     node.apply_result(result);
 }
 
-Chess::Move MCTSTree::best_move() const
+Optional<MCTSTree&> MCTSTree::child_with_this_move(Chess::Move chess_move)
```

Comment:
I don't think we usually refer to arguments with "this", just `child_with_move` seems fine